client/stores/user.ts

```diff
@@ -1,7 +1,11 @@
 import { defineStore } from 'pinia'
+import { LOGIN_PATH, onLoginRoute } from '~/utilities/url'
 
 export type ProfileData = {
-  authenticated: boolean
+  /**
+   * Until we fetch their profile their authentication state is unknown which we'll express as `undefined`
+   */
+  authenticated: undefined | boolean
```

Comment:
This is the main logic change. Until we fetch the profile we don't know whether we're logged in, so there are 3 possible auth states not 2.

Reply:
Seems reasonable to me. I suppose we could squabble about whether undefined or null is the better third state. I'm personally inclined toward null due to a fear of the undefined, but don't feel strongly. I only mention it because we have a null value serving a similar purpose beside it.

I guess this is different in that it's "don't know yet" versus "no value" since id would be null if authenticated were false.